test/core/handshake/client_ssl.c

```diff
@@ -147,20 +152,34 @@ static void server_thread(void *arg) {
   }
 
   // Load key pair.
-  if (SSL_CTX_use_certificate_file(ctx, SSL_CERT_PATH, SSL_FILETYPE_PEM) < 0) {
+  if (SSL_CTX_use_certificate_file(
+          ctx, ((load_ecdsa) ? SSL_EC_CERT_PATH : SSL_RSA_CERT_PATH),
+          SSL_FILETYPE_PEM) < 0) {
+    ERR_print_errors_fp(stderr);
+    abort();
+  }
+  if (SSL_CTX_use_PrivateKey_file(
+          ctx, ((load_ecdsa) ? SSL_EC_KEY_PATH : SSL_RSA_KEY_PATH),
+          SSL_FILETYPE_PEM) < 0) {
     ERR_print_errors_fp(stderr);
     abort();
   }
-  if (SSL_CTX_use_PrivateKey_file(ctx, SSL_KEY_PATH, SSL_FILETYPE_PEM) < 0) {
+
+  // Enable automatic DHE and ECDHE selection
+  if (!SSL_CTX_set_ecdh_auto(ctx, 1)) {
```

Comment:
It's not really needed as you are already calling `SSL_CTX_set_tmp_ecdh` in `populate_ssl_context`. I had to add the call to `SSL_CTX_set_ecdh_auto` for the test to enable it but the main TSI is all set.